test/Driver/linker-clang_rt.swift

```diff
@@ -0,0 +1,36 @@
+// REQUIRES: OS=macosx
```

Comment:
What is this test testing? What is essential vs incidental? Comments to that effect would be helpful.